### PR TITLE
add legacy assets flag to wrangler dev/deploy

### PR DESCRIPTION
**Pull request by emily-shen**
### Summary
Updating docs for wrangler dev/deploy --assets

Screenshot 2024-08-12 at 11 50 47

See [this PR](https://github.com/cloudflare/workers-sdk/pull/6272) for further details.

### CODE REVIEW

File: content/workers/wrangler/commands.md [hunk 1]
```diff
@@ -789,9 +789,12 @@ As of Wrangler v3.2.0, `wrangler dev` is supported by any Linux distributions pr
...
 - `--local-upstream` {{<type>}}string{{</type>}} {{<prop-meta>}}optional{{</prop-meta>}}
   - Host to act as origin in local mode, defaults to `dev.host` or route.
-- `--assets` {{<type>}}string{{</type>}} {{<prop-meta>}}optional{{</prop-meta>}}
+- `--legacy-assets` {{<type>}}string{{</type>}} {{<prop-meta>}}optional{{</prop-meta>}} {{<prop-meta>}}experimental{{</prop-meta>}}
   - Root folder of static assets to be served.
-  - Use in combination with `--name` and `--latest` for basic static file hosting. For example: `wrangler dev --name personal_blog --assets dist/ --latest`.
+  - Use in combination with `--name` and `--latest` for basic static file hosting. For example: `wrangler dev --name personal_blog --legacy-assets dist/ --latest`.
+- `--assets` {{<type>}}string{{</type>}} {{<prop-meta>}}optional{{</prop-meta>}} {{<prop-meta>}}experimental{{</prop-meta>}}
+  - Root folder of static assets to be served.
+  - {{<Aside type="warning">}}This is an experimental feature and its behavior will be changing soon. Use `--legacy-assets` instead.{{</Aside>}}
```
Comment (petebacondarwin) [inline content/workers/wrangler/commands.md]: I don't know what discussions there have been about the copy here, but I think this is a bit ambiguous.
Could we call out that the user should use `legacy-assets` if they want to continue using the current behaviour when the `--assets` behaviour changes.

**Review by CarmenPopoviciu — APPROVED** [review]

File: content/workers/wrangler/commands.md (hunk 1 above)
Comment (CarmenPopoviciu) [reply content/workers/wrangler/commands.md]: But we are already doing that? `Use --legacy-assets instead.` Or is that not specific enough?

Comment (petebacondarwin) [reply content/workers/wrangler/commands.md]: How about something like:

```suggestion
  - {{<Aside type="warning">}}This is an experimental feature and its behavior will be changing soon. Use `--legacy-assets` instead to ensure that you get consistent behaviour when this option changes.{{</Aside>}}
```

**Comment by CarmenPopoviciu** [comment]
@petebacondarwin this LGTG now

**Review by petebacondarwin — APPROVED** [review]

**Pull request merged**